### ISSUE TITLE
unify connection lists

### ISSUE THREAD

**Issue by tzunghaor** [TablePlus/TablePlus#2473]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0): not relevant for this UI issue

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81): Version 4.0.0 (370)

3. The steps to reproduce this issue:  
  I wanted to edit a connection (the server has changed), and open it in the current window, and I found no way to do it.
  a. On the welcome screen connection list there is no search input (when typing characters it jumps to the first match, but the hide non-matching functionality would be useful)
  b. Connection -> Open a Connection: there is no possibility to edit a connection or create a group
  c. File -> New Workspace: this brings up the welcome screen, and has no possibility to open connection in same window.
  d. I can open the welcome dialog by Connection/New..., then closing the new connection wizard.

Main Suggestion: I would suggest to have the same UI element on the welcome screen and in the Open a Connection window, which would have both the filter-by-search-input and possibility to edit existing connections.

Suggestion B: If the File/New Workspace, Connection/Open a Connection... (which can already open connection in a new workspace) and Connection/New... would lead to basically the same dialog, then it would be enough to have one menu item for these (e.g. Connection/Connections...)

**Comment by huyphams** [comment]
a. On the welcome screen connection list there is no search input (when typing characters it jumps to the first match, but the hide non-matching functionality would be useful)
=> You can hit command + F TablePlus will show you the search box.

b. Connection -> Open a Connection: there is no possibility to edit a connection or create a group
=> Right click on the welcome screen or in the group/connection to edit it.

c. File -> New Workspace: this brings up the welcome screen, and has no possibility to open connection in same window.
=> Command + shift + K and Command + K should do the trick


d. I can open the welcome dialog by Connection/New..., then closing the new connection wizard.
I don't understand could you explain it?

**Comment by huyphams** [comment]
All those features are available so I will close this issue. Please let me know if you have any questions.

**Comment by tzunghaor** [comment]
Wow, thanks for the quick response. I didn't know that command + F brings up the search box. Would it be possible to display it there by default too? I find it super useful, and since it is visible in the other connection list, but not in the welcome, I (and probably others) thought that it is simply not available there.

Yes, all these functionality is accessible one way or the other, but I, as a mere user I see no advantage in having a dialog where I cannot edit/create connections, and another where I cannot open the connection in the same window. For me it is just confusing.

> I wanted to edit a connection (the server has changed), and open it in the current window, and I found no way to do it.

Sorry, I wrote that wrong, I found the way (edit in the welcome dialog, then use the Open a Connection dialog to open it), but it is more cumbersome as I'd expect. I would find the workflow easier if I could do it in a single dialog.

> d. I can open the welcome dialog by Connection/New..., then closing the new connection wizard.

It's just that I would expect, if I open the new connection dialog directly from the workspace, then it opens only the new connection dialog, not additionally the welcome dialog. But it's no biggie.

---

All in all I think it would give a better user experience to have just a single connection list UI element - not necessary a single dialog, I mean only that the list widget could provide the same functionality everywhere.

But I agree that are more important open issues, I wrote this down however to not go forgotten.
